### PR TITLE
Add slugs locale orderedset

### PR DESCRIPTION
**Pull request by HawkiesZA**
## Purpose
The migration fails on QA because we're not setting a default locale for ordered sets.

## Solution
This sets a default locale for ordered sets and also fixes the slug deduplication (which did not take into account that all slugs will be empty at the start).

#### Checklist
- [x] Added or updated unit tests
- [ ] Added to release notes
- [ ] Updated readme/documentation (if necessary)

### CODE REVIEW

File: home/migrations/0085_orderedcontentset_locale_orderedcontentset_slug.py [hunk 1]
```diff
@@ -3,47 +3,53 @@
...
-            candidate_slug = slug
-            while OrderedContentSet.objects.filter(
-                slug=candidate_slug, locale=locale
-            ).exists():
+    for slug in duplicate_slugs:
+        ordered_content_sets = OrderedContentSet.objects.filter(slug=slug)
+        for ordered_content_set in ordered_content_sets:
+            slug_from_name = (
+                ordered_content_set.name.lower().replace(" ", "-").replace("_", "-")
+            )
```
Comment (rudigiesler) [inline home/migrations/0085_orderedcontentset_locale_orderedcontentset_slug.py:22]: Maybe we can use Django's `slugify` function here? https://docs.djangoproject.com/en/5.1/ref/utils/#django.utils.text.slugify . I think slugs are allowed to have underscores

File: home/tests/test_migrations.py [hunk 2]
```diff
@@ -53,32 +53,22 @@ def test_deduplicates_orderedcontentset_slugs(self) -> None:
...
-        )
-        OrderedContentSet.objects.create(
-            name="unique", slug="unique", locale=default_locale
-        )
+        # Create ordered content sets with blank slugs
+        OrderedContentSet.objects.create(name="first", slug="", locale=default_locale)
+        OrderedContentSet.objects.create(name="first", slug="", locale=default_locale)
+        OrderedContentSet.objects.create(name="first_pt", slug="", locale=pt_locale)
+        OrderedContentSet.objects.create(name="third", slug="", locale=default_locale)
+        OrderedContentSet.objects.create(name="unique", slug="", locale=default_locale)
```
Comment (rudigiesler) [inline home/tests/test_migrations.py:61]: Maybe we can have some of the names here with whitespace and repeated dashes, to ensure that we're converting name -> slug correctly?

**Review by rudigiesler — COMMENTED** [review]
Just a small thing with how we convert name to slug that I think we can improve, otherwise looks good!

File: home/migrations/0085_orderedcontentset_locale_orderedcontentset_slug.py (hunk 1 above)
Comment (HawkiesZA) [reply home/migrations/0085_orderedcontentset_locale_orderedcontentset_slug.py:22]: Didn't know that existed. Thanks!

File: home/tests/test_migrations.py (hunk 2 above)
Comment (jerith) [reply home/tests/test_migrations.py:61]: And mixed case?

Comment (HawkiesZA) [reply home/tests/test_migrations.py]: slugify lowercases

**Review by rudigiesler — APPROVED** [review]

**Pull request merged**